src/libstd/sys/unix/thread.rs

```diff
@@ -314,21 +316,30 @@ pub fn sleep(dur: Duration) {
 // is created in an application with big thread-local storage requirements.
 // See #6233 for rationale and details.
 //
-// Link weakly to the symbol for compatibility with older versions of glibc.
-// Assumes that we've been dynamically linked to libpthread but that is
-// currently always the case.  Note that you need to check that the symbol
-// is non-null before calling it!
+// Use dlsym to get the symbol value at runtime, for compatibility
+// with older versions of glibc.  Assumes that we've been dynamically
+// linked to libpthread but that is currently always the case.  We
+// previously used weak linkage (under the same assumption), but that
+// caused Debian to detect an unnecessarily strict versioned
+// dependency on libc6 (#23628).
 #[cfg(target_os = "linux")]
 fn min_stack_size(attr: *const libc::pthread_attr_t) -> libc::size_t {
     type F = unsafe extern "C" fn(*const libc::pthread_attr_t) -> libc::size_t;
-    extern {
-        #[linkage = "extern_weak"]
```

Comment:
I’m not a Mac user; does Rust use glibc on Mac? I figured we shouldn’t waste time looking up `__pthread_get_minstack` on non-glibc.